docker/docker+.bsh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+set -eu
+
+if [[ $# == 0 ]]; then
+  FILES=(-)
+else
+  FILES=("${@}")
+fi
+
+CUR_DIR=$(basename ${BASH_SOURCE[0]})
+
+for FILE in "${FILES[@]}"; do
+  if [ "${FILE}" == "-" ]; then
+    FILE=/dev/stdin
+  fi
+  while IFS='' read -r LINE || [ -n "${LINE}" ]; do
+
+    #My template substitution
+    while [[ "$LINE" =~ (\[\{)([A-Za-z0-9_]*)(\}\]) ]]; do
+      LINE=${LINE/"${BASH_REMATCH[0]}"/${!BASH_REMATCH[2]-}}
+    done
+    
+    #My SOURCE substitution
+    if echo $LINE | grep -q ^SOURCE; then
+      SOURCE=$(echo "${LINE}" | sed -r 's|^SOURCE\s+(\S*)\s*|\1|')
```

Comment:
This line blows up on the mac, which apparently has a really old version of sed. However, using `-E` doesn't seem to work either:

```
$ echo "SOURCE bootstrap_centos_5.dockerfile" | sed -E 's|^SOURCE\s+(\S*)\s*|\1|'
SOURCE bootstrap_centos_5.dockerfile
```

Pretty sure the script wants just `bootstrap_centos_5.dockerfile` there.